franklin/github/tests.py

```diff
@@ -39,3 +41,44 @@ def test_get_franklin_config(self, mock_get):
             self.assertEqual(franklin_config.get('hello', None), 'world')
         else:
             self.fail('Received an HTTP Response instead of data from method')
+
+
+class OauthTokenTestCase(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(username="testuser", password="a")
+        social = self.user.social_auth.create(provider='github')
+        social.extra_data['access_token'] = ''
+        social.save()
+        self.owner = Owner.objects.create(
+            name='istrategylabs', github_id=607333)
+        self.site = Site.objects.create(
+            owner=self.owner, name='franklin-dashboard', github_id=45864453)
+
+    @mock.patch('github.views.do_auth')
+    @mock.patch('github.views.make_rest_post_call')
+    def test_get_franklin_config(self, mock_post, mock_auth):
+        """ Tests the method to fetch an oauth token
+        """
+        # mocking a request object received by the API
+        request = self.factory.post('/auth/github')
+        request.data = {'code': 'asdf',
+                        'clientId': 'asdf',
+                        'redirectUrl': 'asdf'}
```

Comment:
Using `RequestFactory`, we can now mock a GET/POST call coming to API. This means we can start creating unit tests that exercise all of the API endpoints directly without spinning up a server or anything like that.